main/src/ca/uwaterloo/flix/language/ast/Name.scala

```diff
@@ -26,31 +26,25 @@ object Name {
   /**
     * Returns the given string `fqn` as a qualified name.
     */
-  def mkQName(fqn: String, sp1: SourcePosition = SourcePosition.Unknown, sp2: SourcePosition = SourcePosition.Unknown): QName = {
+  def mkQName(fqn: String, loc: SourceLocation = SourceLocation.Unknown): QName = {
     if (!fqn.contains('.'))
-      return QName(sp1, Name.RootNS, Ident(sp1, fqn, sp2), sp2)
+      return QName(Name.RootNS, Ident(loc.sp1, fqn, loc.sp2), loc)
 
     val index = fqn.indexOf('.')
     val parts = fqn.substring(0, index).split('/').toList
     val name = fqn.substring(index + 1, fqn.length)
-    mkQName(parts, name, sp1, sp2)
+    mkQName(parts, name, loc)
   }
 
   /**
     * Creates a qualified name from the given namespace `ns` and name `name`.
     */
-  def mkQName(ns: List[String], name: String, sp1: SourcePosition, sp2: SourcePosition): QName = {
-    val loc = SourceLocation(isReal = true, sp1, sp2)
-    val nname = NName(ns.map(t => Name.Ident(sp1, t, sp2)), loc)
-    val ident = Ident(sp1, name, sp2)
-    QName(sp1, nname, ident, sp2)
+  def mkQName(ns: List[String], name: String, loc: SourceLocation): QName = {
+    val nname = NName(ns.map(t => Name.Ident(loc.sp1, t, loc.sp2)), loc)
+    val ident = Ident(loc.sp1, name, loc.sp2)
+    QName(nname, ident, loc)
   }
 
-  /**
-    * Returns the given identifier `ident` as qualified name in the root namespace.
-    */
-  def mkQName(ident: Ident): QName = QName(ident.sp1, RootNS, ident, ident.sp2)
```

Comment:
Why did we get rid of this helper? Seems like it's used a lot.

Reply:
My plan is to get rid of these smart constructors and focus _on the data_.

Reply:
We could an expression to deal with that.

Reply:
IMO the root of the problem is that our standard library is unqualified. We should just have a `Flix` namespace and put the who stdlib in it.

Reply:
Then we can fully separate qualified and unqualified names

Reply:
This is orthogonal though

Reply:
Ok maybe that's actually orthogonal, but we should do it anyway :)